### DISCUSSION TITLE
Embree Patch Unexpected End of File

### DISCUSSION THREAD

**Discussion by RickShahid** [dreamworksanimation/openmoonray#105] [Q&A]
Hi, I am trying to build MoonRay on Rocky Linux v8.6 and I am running into the following error when calling "cmake --build . -- -j 64"

[ 73%] Performing patch step for 'embree'
patching file include/embree3/rtcore_common.isph
patch unexpectedly ends in middle of line
Reversed (or previously applied) patch detected!  Assume -R? [n]
Apply anyway? [n]
Skipping patch.
1 out of 1 hunk ignored -- saving rejects to file include/embree3/rtcore_common.isph.rej
gmake[2]: *** [CMakeFiles/embree.dir/build.make:117: embree-prefix/src/embree-stamp/embree-patch] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:455: CMakeFiles/embree.dir/all] Error 2
gmake: *** [Makefile:91: all] Error 2

Any ideas / suggestions?

When I look at the source file at https://github.com/dreamworksanimation/openmoonray/blob/release/building/embree_missing_type.patch, it does look to be an incomplete file. Why is that?

Thanks.

Rick

**Comment by rgwilson68** [comment]
I think the file is complete : it lists 3 lines of context before and after the change. 
The message "Reversed (or previously applied) patch detected!" makes me think the source might already have the patch applied ... are you definitely downloading version v3.12.1 of embree?
Rocky Linux might have a different version of 'patch' : in Centos-7 "patch -version" shows "GNU patch 2.7.1". I don't know why  a different version would fail though...

**Comment by RickShahid** [comment]
I just ran another build test and the embree patch issue did not occur. I am not sure what changed. I must have had an issue in my build directory previously. In any case, I am able to move forward now. For reference, the version of patch that gets installed on Rocky Linux v8.6 is "GNU Patch 2.7.6"
